Description: Do not add drive 'boot=on' param when a kernel is specified
Author: Jim Fehlig <jfehlig@novell.com>
Origin: upstream
Bug-Ubuntu: https://bugs.launchpad.net/bugs/720426

libvirt-tck was failing several domain tests [1] with qemu 0.14, which
is now less tolerable of specifying 2 bootroms with the same boot index [2].

Drop the 'boot=on' param if kernel has been specfied.

[1] https://www.redhat.com/archives/libvir-list/2011-February/msg00559.html
[2] http://lists.nongnu.org/archive/html/qemu-devel/2011-02/msg01892.html
---
 src/qemu/qemu_command.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: libvirt-0.8.7/src/qemu/qemu_command.c
===================================================================
--- libvirt-0.8.7.orig/src/qemu/qemu_command.c	2011-02-18 09:45:24.516201412 -0600
+++ libvirt-0.8.7/src/qemu/qemu_command.c	2011-02-18 09:45:24.564201220 -0600
@@ -2984,7 +2984,7 @@
         int bootCD = 0, bootFloppy = 0, bootDisk = 0;
 
         /* If QEMU supports boot=on for -drive param... */
-        if (qemuCmdFlags & QEMUD_CMD_FLAG_DRIVE_BOOT) {
+        if (qemuCmdFlags & QEMUD_CMD_FLAG_DRIVE_BOOT && !def->os.kernel) {
             for (i = 0 ; i < def->os.nBootDevs ; i++) {
                 switch (def->os.bootDevs[i]) {
                 case VIR_DOMAIN_BOOT_CDROM:
